app/controllers/referee_interface/reference_controller.rb

```diff
@@ -0,0 +1,47 @@
+module RefereeInterface
+  class ReferenceController < ActionController::Base
+    layout 'application'
+
+    def feedback
+      reference = Reference.find_by(token: params[:token])
+
+      if reference.present?
+        @reference_form = RefereeInterface::ReferenceForm.build_from_reference(reference)
+        @application = reference.application_form
+      else
+        render_404
+      end
+    end
+
+    def submit_feedback
+      reference = Reference.find_by(token: params[:token])
```

Comment:
This doesn't check if the token exists before doing the work, using `find_by!` would help. 

I also wonder what happens if the token is empty, since it'll match the `Reference` that currently don't have a `token` set.